crypto/bls_thresholdsign.go

```diff
@@ -572,9 +572,12 @@ func BLSThresholdKeyGen(size int, threshold int, seed []byte) ([]PrivateKey,
 	}
 	// Generate a polynomial P in Zr[X] of degree t
 	a := make([]scalar, threshold+1)
-	randZrStar(&a[0])
-	for i := 1; i < threshold+1; i++ {
-		randZr(&a[i])
+	randZrStar(&a[0]) // non-identity key
+	if threshold > 0 {
+		for i := 1; i < threshold+1; i++ {
+			randZrStar(&a[i])
+		}
+		randZrStar(&a[threshold]) // enforce the polynomial degree
```

Comment:
It seems like we are generating keys 0 and 1-n separately, and generating key n twice? I'm not sure why it's structured this way, rather than simply:
```suggestion
	for i := 0; i < threshold+1; i++ {
		randZrStar(&a[i])
	}
```

Reply:
Thanks for pointing this. There are two issues/typos:
 - the `a_i` inside the loop don't have to be non-zero, so `randZr` is enough (that was my intention after separating `a_0` and `a_t`)
 - the loop should stop at `threshold` to not generate twice.